Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindowManager.createHomeDockIntent:COMMENT
Method Modifier: internal    
Comment:/**
 * Return an Intent to launch the currently active dock app as home.  Returns
 * null if the standard home should be launched, which is the case if any of the following is
 * true:
 * <ul>
 * <li>The device is not in either car mode or desk mode
 * <li>The device is in car mode but ENABLE_CAR_DOCK_HOME_CAPTURE is false
 * <li>The device is in desk mode but ENABLE_DESK_DOCK_HOME_CAPTURE is false
 * <li>The device is in car mode but there's no CAR_DOCK app with METADATA_DOCK_HOME
 * <li>The device is in desk mode but there's no DESK_DOCK app with METADATA_DOCK_HOME
 * </ul>
 * @return A dock intent.
 */

@@ -9,14 +9,17 @@
         if (ENABLE_DESK_DOCK_HOME_CAPTURE) {
             intent = mDeskDockIntent;
         }
+    } else if (mUiMode == Configuration.UI_MODE_TYPE_WATCH && (mDockMode == Intent.EXTRA_DOCK_STATE_DESK || mDockMode == Intent.EXTRA_DOCK_STATE_HE_DESK || mDockMode == Intent.EXTRA_DOCK_STATE_LE_DESK)) {
+        // Always launch dock home from home when watch is docked, if it exists.
+        intent = mDeskDockIntent;
     }
     if (intent == null) {
         return null;

